### PR TITLE
[4.0] blog intro image

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes
This PR fixes multiple issues with the blog_intro_image layout. If linked titles is ON then the image is also made into a link. 

- Accessibility error :  Identical links next to each other
- Accessibility / html error : No link name on the image link ( They are just announced by a screenreader as link without any indication of what they are a link to)
- Structured data error : missing itemprop on the image link
- Authentication issue : image link did not support/check the show_noauth param

### Expected result AFTER applying this Pull Request
After applying this PR these issues are all resolved when the option is enabled.
There is 
- a new option in the global blog layout options "Link Intro Image" - the default is NO (because the default for linked article title is yes)
- a new option in the blog layout menu as above
- a new option in the featured layout menu as above

### Documentation Changes Required
None

### CODE REVIEW

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 004c75db21ed0169c73cfc313bdfafbbea79f571<br /><br />All seemed to work as described. I don't know about 'the show_noauth param' thing - never met it.

On documentation: We (I) need to update the Help screens when this gets committed. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30823">issues.joomla.org/tracker/joomla-cms/30823</a>.</sub>

**Comment by chmst** [comment]
All works as expected and is better than before.  
An accessibility error could be: "redundant link", as there are two (or if there is a readmore three) equal links are close together. 

I give a positive test because it is better than before and users must know what they do.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 004c75db21ed0169c73cfc313bdfafbbea79f571<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30823">issues.joomla.org/tracker/joomla-cms/30823</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30823">issues.joomla.org/tracker/joomla-cms/30823</a>.</sub>

**Comment by HLeithner** [comment]
Do you really thing it's needed to add a new option? I mean we try to reduce useless options and this looks like a template option and not a com_content option in my opinion...

**Comment by brianteeman** [comment]
No its not useless!! What is useless are the current options which are not usable by a regular user. This makes them usable.

**Comment by HLeithner** [comment]
It's not useless because it's not useless is not really a good answer ;-)

but ok if you think it's needed to disable the link on an images (which should always be on for people which have troubles to target the title correctly or always click on the image first because it's bigger like me) then I'm ok with it.

**Comment by brianteeman** [comment]
Either we are serious about creating a cms that will create accessible content or we are not.

**Comment by HLeithner** [comment]
why is the link on an image a accessible problem? (sorry that I don't know it)

I'm happy with the rest of the pr

**Comment by brianteeman** [comment]
It is stated in the description

Accessibility error : Identical links next to each other
Accessibility / html error : No link name on the image link ( They are just announced by a screenreader as link without any indication of what they are a link to)

**Comment by HLeithner** [comment]
> Accessibility / html error : No link name on the image link ( They are just announced by a screenreader as link without any indication of what they are a link to)

that's the other part of the pr that's should be fixed and is fine I think.

> Accessibility error : Identical links next to each other

That's something I can't understand why it is a problem and by the way if I don't understand who should someone understand it who only want's to have a website? How should anyone think that this options is a a11y option?

can one of the links automatically be disabled for a screenreader?

**Comment by brianteeman** [comment]
OK Let me try to explain the multiple issues.

Firstly if you select linked titles you also get linked image. You didnt ask for that but you got it anyway. The link on the image was created without any meaningful description. This means that a screenreader would just announce "Link" for every linked image. As this is for the blog intro image then you will now have multiple links on the page that all have the same name "Link". If that is fixed by ensuring the link has a meaningful name then you hit the second problem. You now have two identical links next to each other. This is against the WCAG rules but it also means a user using any form of assistive technology has to navigate past double the number of links to  get to the one that they want.

This is not about making accessibility an option or something that you have to configure. This is about making all content accessible. 

In my JaB talk I talked about how we have been led down the path of responsive design where we ensured that the content was great on any device. So that our websites worked on all devices without doing anything special. This device first approach completely ignored people. A people first approach ensures that content works great for all people without doing anything special.


> can one of the links automatically be disabled for a screenreader?

Not just about screenreaders. Anyone navigating a site in a linear manner eg with a keyboard also has the problem of double identical links. And no you can't do that  and nor should you. 

If you want to see a really typical example of why this needs to be fixed just check out the supposedly accessible template https://paswjoomla.net/joomla/

**Comment by laoneo** [comment]
But should then the CMS not come with a proper default and when somebody want's to change the default behavior, then make a template override? I understand @HLeithner that these tiny options are flooding the core already.

**Comment by chmst** [comment]
A correct default solution would be to generate the image without link.  As @laoneo says. This was default until someone has made a PR for linking the image too. 
But I know that, if we remove the link on the image, we soon will have a PR for linking the image  - because it seems to be a natural human behaviour to klick on images in a blog, more than on headlines. 

I understand @HLeithner and would like to reduce the number of params instead of increasing them. And I am sure that all users will choose to link image AND title because they don't know why this is not a11y (or don't care about a11y). 

So this PR makes sure that  if there is a link on the image, it at least is correct.

**Comment by ghost** [comment]
> I am sure that all users will choose to link image AND title because they don't know why this is not a11y (or don't care about a11y)

sure too

**Comment by brianteeman** [comment]
We really should be making it easier for content creators to produce accessible content. Not just for the joomla UI to be accessible. If I had my preference it would be to only allow either title or image to be linked eg
Blog links ->
- none
- readmore
- title
- image

But I was pretty sure that would get rejected so I made it an option

**Comment by brianteeman** [comment]
>> I am sure that all users will choose to link image AND title because they don't know why this is not a11y (or don't care about a11y)

> sure too

thats why I proposed the accessibility checker

**Comment by laoneo** [comment]
Linking the image was done in 2014, so the web changed quite a bit since then. I think it is fine to get rid of the links. For reference, here is the pr which added them https://github.com/joomla/joomla-cms/pull/4134.

**Comment by HLeithner** [comment]
@brianteeman please simple remove the link on the image since it is a a11y problem we shouldn't confuse people with options they shouldn't use^^ if someone want's the image linked he/she should do a layout override.

And please add a comment in the source why the image doesn't have a link anymore.

**Comment by brianteeman** [comment]
There is nothing wrong with having the image as a link (when the code is correct). The problem is only when you have a linked image AND a linked title AND they are next to each other. So no I wont remove it. Either accept or reject this PR. It is correct as it is.

**Comment by laoneo** [comment]
Can we not detect that programmatically and create then proper HTML code? Is it really necessary that a human needs to define here how the output should be that it is accessible? With my little knowledge I have the impression this is something core should do correct out of the box.

**Comment by brianteeman** [comment]
You're a better person than me then if you can write the logic to determine this programmatically with the current options

With the proposed defaults core will do it correctly out of the box.

**Comment by drmenzelit** [comment]
I like this approach
https://inclusive-components.design/cards/
to make blog elements clickable. 
I have it working on some pages using overrides.
Maybe something to think about.

**Comment by brianteeman** [comment]
Thanks @rdeutz

File: installation/sql/postgresql/base.sql [hunk 1]
```diff
@@ -166,7 +166,8 @@ INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder",
...
+"link_intro_image":"0","show_pagination_limit":"1","filter_field":"hide","show_headings":"1","list_show_date":"0","date_format":"","list_show_hits":"1","list_show_author":"1","orderby_pri":"order","orderby_sec":"rdate","order_date":"published","show_pagination":"2","show_pagination_results":"1","show_feed_link":"1","feed_summary":"0"}', '', 0, 0),
```
Comment (richard67) [inline installation/sql/postgresql/base.sql:170]: This new line within the params string is wrong. Will make PR to fix it.

Comment (brianteeman) [reply installation/sql/postgresql/base.sql:170]: oops - surprised we dont have tests for this

Comment (richard67) [reply installation/sql/postgresql/base.sql:170]: Well it doesn't cause an SQL error, so the installation at the system test for postgresql doesn't fail, and SQL code style we don't test.

Comment (richard67) [reply installation/sql/postgresql/base.sql:170]: Thanks for testing the PR.

**Pull request merged**